unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Avatar.cs

```diff
@@ -21,13 +21,14 @@ public class Avatar : IAvatar
         private readonly IGPUSkinning gpuSkinning;
         private readonly IGPUSkinningThrottler gpuSkinningThrottler;
         private readonly IEmoteAnimationEquipper emoteAnimationEquipper;
+        private readonly IBaseAvatar baseAvatar;
         private CancellationTokenSource disposeCts = new CancellationTokenSource();
 
         public IAvatar.Status status { get; private set; } = IAvatar.Status.Idle;
         public Vector3 extents { get; private set; }
         public int lodLevel => lod?.lodIndex ?? 0;
 
-        public Avatar(IAvatarCurator avatarCurator, ILoader loader, IAnimator animator, IVisibility visibility, ILOD lod, IGPUSkinning gpuSkinning, IGPUSkinningThrottler gpuSkinningThrottler, IEmoteAnimationEquipper emoteAnimationEquipper)
+        public Avatar(IAvatarCurator avatarCurator, ILoader loader, IAnimator animator, IVisibility visibility, ILOD lod, IGPUSkinning gpuSkinning, IGPUSkinningThrottler gpuSkinningThrottler, IEmoteAnimationEquipper emoteAnimationEquipper, IBaseAvatar baseAvatar)
```

Comment:
I would change the signature of the method and put the `BaseAvatar` first to give a bit of visibility of the order of execution within the Avatar.